packages/grid/x-data-grid/src/DataGrid/DataGrid.tsx

```diff
@@ -595,6 +578,13 @@ DataGridRaw.propTypes = {
    * @default "client"
    */
   paginationMode: PropTypes.oneOf(['client', 'server']),
+  /**
+   * The pagination model of type [[GridPaginationModel]] which refers to current `page` and `pageSize`.
+   */
+  paginationModel: PropTypes.shape({
+    page: PropTypes.number.isRequired,
+    pageSize: PropTypes.number.isRequired,
```

Comment:
I'm wondering if it makes sense to simplify the name to just `size` given the context is set for pagination. 🤔
```suggestion
    size: PropTypes.number.isRequired,
```

Reply:
Yeah may be we can consider doing that.

Another thing, I was also thinking to rename `rowsPerPageOptions` to `pageSizeOptions` as I have to sometimes think twice to reach the conclusion that `pageSize` and `rowPerPageOptions` are related.
`pageSize` and `pageSizeOptions` look more logical.

If we go with paginationModel.size it could be
```tsx
paginationModel: {
  page: number;
  size: number
}
```

And the `rowsPerPageOptions` could be `paginationOptions` or `pageSizeOptions`

Reply:
@DanailH I'm not sure saving 4 characters worth decreasing clarity. 

`paginationModel.pageSize` is obviously, it's the number of items per page
`paginationModel.size` could be either the number of items per page or the number of pages

Reply:
Fair enough, assuming there can be more numbers that represent different things.